feat: create IAM user for pipeline ECR tasks

patheard opened the pull request:
This creates our ECR user, which we'll use in the `covid-alert-metrics` pipeline change.

It also:
* Fixes a bug with the `deploy-containers` workflow aws-actions (bumps to latest for both), and
* Fixes the `Check Success` step to catch PR workflows with failing `tf fmt` steps.
* ~~Makes the PR `terraform` workflow hard fail on a `tf fmt -check` because I completely missed it failing in my last PR~~ 

Fixes: #34 
Fixes: #38
Related: cds-snc/covid-alert-metrics#68

patheard committed:
chore: bump aws-action versions in workflow

Updated to latest version of both actions as they only include
dependency updates and one new feature, which defaults to
using the same behaviour as what we currently have.

Changes introduced by each bump:
https://github.com/aws-actions/amazon-ecs-render-task-definition/compare/v1.0.10...v1.0.20
https://github.com/aws-actions/amazon-ecs-deploy-task-definition/compare/v1.3.5...v1.4.4

patheard committed:
feat: cause TF fmt to fail PR workflow

Since the TF apply workflow hard fails on this, it makes sense to have
this caught early in the PR workflow as well.

patheard committed:
feat: create IAM user for pipeline ECR tasks

This user is almost full control over an ECR but
cannot delete the ECR itself.

CalvinRodo commented on .github/workflows/terraform.yml:
```diff
@@ -56,12 +56,10 @@ jobs:
       - name: Terraform Format
         id: fmt
         run: terraform fmt -check -no-color
-        continue-on-error: true
```
Put this back in and fix check_success.

CalvinRodo reviewed (changes requested):
Make changes

patheard committed:
fix: Check Success step tf fmt exit code condition

Terraform fmt returns a non-zero exit code equal to the number of
files that were not properly formatted.

patheard committed:
fix: add the continue-on-error to the correct step

Accidentally added it to the tf validate step.

CalvinRodo reviewed (approved):
:taco::cat:
